### PR TITLE
Add getCategoriesByTag

### PR DESCRIPTION
**Pull request by BasileNouvellet**
Add method to get all Categories possessing a given tag name
Should be similar to getPoisByTag method

### CODE REVIEW

File: packages/adsum-utils/services/ClientAPI.js [hunk 1]
```diff
@@ -123,6 +123,19 @@ class ClientAPI {
         return this.entityManager.getRepository('Category').findBy(filter);
     }
 
+    getCategoriesByTag(tagName) {
+        let categoriesToReturn = [];
```
Comment (moroine) [inline packages/adsum-utils/services/ClientAPI.js]: rename to `result`

File: packages/adsum-utils/services/ClientAPI.js [hunk 2]
```diff
@@ -123,6 +123,19 @@ class ClientAPI {
...
+        const tags = this.getTagBy({name: tagName});
+
+        for (let tag of tags) {
+            for (let category of tag.categories.values) {
+                const categoryId = category.value;
+                categoriesToReturn.push(this.getCategory(categoryId));
```
Comment (moroine) [inline packages/adsum-utils/services/ClientAPI.js]: Prefer `categoriesToReturn.push(... this.getCategories(tag.categories));`

File: packages/adsum-utils/services/ClientAPI.js [hunk 3]
```diff
@@ -123,6 +123,19 @@ class ClientAPI {
...
+            }
+        }
+        return categoriesToReturn;
```
Comment (moroine) [inline packages/adsum-utils/services/ClientAPI.js]: Need to make the result unique, if a Category has two tags then it will be duplicated...

Comment (BasileNouvellet) [reply packages/adsum-utils/services/ClientAPI.js]: A Category can have 2 tags with the same name?

**Pull request merged**